lib/internal/fs/glob.js

```diff
@@ -268,7 +269,7 @@ class Glob {
   constructor(pattern, options = kEmptyObject) {
     validateObject(options, 'options');
     const { exclude, cwd, withFileTypes } = options;
-    this.#root = cwd ?? '.';
```

Comment:
Nit: These additions make optimizations extremely hard. We should avoid adding them.

Reply:
You're commenting on a deleted line, it's not very clear what you mean

Reply:
Adding a URL support to input field